install-deps.sh

```diff
@@ -312,7 +312,7 @@ function preload_wheels_for_tox() {
     if test "$require" && ! test -d wheelhouse ; then
         type python3 > /dev/null 2>&1 || continue
         activate_virtualenv $top_srcdir || exit 1
-        python3 -m pip install --upgrade pip
+        # python3 -m pip install --upgrade pip
```

Comment:
> Maybe pinning it to `22.02` rather than stopping it?

Do we know what changed after that release?

> Also when newer python versions are applied to project, we might need new pip to pull in the new dependencies which might be unsupported by older pip versions.

Exactly.  This is why I like https://github.com/ceph/ceph/pull/52544 better.

Reply:
> This is why I like #52544 better.

It also gets rid of "fast-deps has no effect when used with the legacy resolver" warnings.

Reply:
> Do we know what changed after that release?

I couldn't find why this changed suddenly. But looks like there have been so many deprecations or changes happening with the legacy versions on some of the [latest pip releases](https://pip.pypa.io/en/stable/news/#deprecations-and-removals)

Reply:
I also prefer to fix this problem for the newest pip, but we may also want to pin it to a specific major version (something like `>=23,<24` IIRC) so that as long as pip is well behaved and don't make big behavior changes in minor versions we get the benefit of taking fixes but not getting surprised by major version changes. It just requires some discipline to periodically check that there's a new major version and then update+test.

Reply:
Here it broke with a minor version: 23.2 was released five days ago, the issue popped up immediately after that.   I didn't dig deeper but it seems to be to close to be a coincidence.

Reply:
OK, I missed that. I had thought the behavior change was going from 22.x to 23.x, thanks for the update.